qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+        """
+        gets absolution position of the motors from the defined origin
+        """
+        result = dict()
+        data = self.ask("PA ?,?,?").split(" ")
+        result["A"] = int(data[0][:-1])
+        result["B"] = int(data[1][:-1])
+        result["C"] = int(data[2])
+
+        return result
+
+    def end_program(self) -> None:
+        """
+        ends the program
+        """
+        self.write("EN")
+
+    def define_position_as_origin(self) -> None:
+        """
+        defines current motors position as origin
+        """
+        self.write("DP 0,0,0")
+
+    def tell_error(self) -> str:
+        """
+        reads error
+        """
+        return self.ask("TC1")
+
+    def stop(self) -> None:
+        """
+        stop the motion of all motors
+        """
+        self.write("ST")
+
+    def abort(self) -> None:
+        """
+        aborts motion and the program operation
+        """
+        self.write("AB")
+
+    def motors_off(self) -> None:
+        """
+        turn all motors off
+        """
+        self.write("MO")
+
+    def begin_motors(self) -> None:
+        """
+        begin motion of all motors simultaneously
+        """
+        self.write("BG")
+
+    def wait_till_motion_complete(self) -> None:
+        """
+        this method waits for the motion on all motors to complete
+        """
+        try:
+            while (
+                int(float(self.ask("MG _BGA")))
+                or int(float(self.ask("MG _BGB")))
+                or int(float(self.ask("MG _BGC")))
+            ):
+                pass
+        except KeyboardInterrupt:
+            self.abort()
+            self.motors_off()
+
+
+class Arm:
+    """
+    Module to control probe arm
+    """
+
+    def __init__(self, controller: DMC4133Controller) -> None:
+        """
+        Initializes the arm class
+
+        Args:
+            controller: an instance of DMC4133Controller
+        """
+        self.controller = controller
+
+        # initialization
+        self.left_bottom_position: Tuple[int, int, int]
+        self.left_top_position: Tuple[int, int, int]
+        self.right_top_position: Tuple[int, int, int]
+
+        # motion directions
+        self._a: np.ndarray  # right_top - left_bottom
+        self._b: np.ndarray  # left_top - left_bottom
+        self._c: np.ndarray  # right_top - left_top
+        self._n: np.ndarray
+        self.norm_a: float
+        self.norm_b: float
+        self.norm_c: float
+
+        # eqn of the chip plane
+        self._plane_eqn: np.ndarray
+
+        # current vars
+        self.current_row: Optional[int] = None
+        self.current_pad: Optional[int] = None
+
+        # chip details
+        self.rows: int
+        self.pads: int
+        self.inter_row_dis: float
+        self.inter_pad_dis: float
+
+        # arm kinematics
+        self.speed: int
+        self.acceleration: int
+        self.deceleration: int
```

Comment:
what is the intention here - should these attrbiutes be public and thus easily changable or should the user be only able to set them via the set arm kinematics method?